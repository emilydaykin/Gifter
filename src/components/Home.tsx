import { FC } from 'react';
import Categories from './categories/Categories';
import Footer from './Footer';

const Home: FC = () => {
  return (
    <>
      <Categories />
      <Footer />
    </>
  );
};

export default Home;
