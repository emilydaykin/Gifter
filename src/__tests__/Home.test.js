import { render, waitFor, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import Home from '../components/Home';

test('Gift categories are rendered', async () => {
  render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );

  expect.assertions(6);

  // waitFor better than wrapping the render with act()
  await waitFor(() => {
    const xmasCategory = screen.getByText(/christmas/i);
    expect(xmasCategory).toBeInTheDocument();
  });

  await waitFor(() => {
    const bdayCategory = screen.getByText(/birthday/i);
    expect(bdayCategory).toBeInTheDocument();
  });

  await waitFor(() => {
    const annivCategory = screen.getByText(/anniversary/i);
    expect(annivCategory).toBeInTheDocument();
  });

  await waitFor(() => {
    const tyCategory = screen.getByText(/thank you/i);
    expect(tyCategory).toBeInTheDocument();
  });

  await waitFor(() => {
    const weddingCategory = screen.getByText(/wedding/i);
    expect(weddingCategory).toBeInTheDocument();
  });

  const browseNowText = screen.queryAllByText(
    (content, element) =>
      element.tagName.toLowerCase() === 'p' &&
      element.innerHTML.toLowerCase().includes('browse now')
  );
  expect(browseNowText).toHaveLength(5);
});

test('Assert Footer is displayed on Home Page', () => {
  render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>
  );

  expect.assertions(1);

  const footer = screen.getByText(/copyright/i);
  expect(footer).toBeInTheDocument();
});
