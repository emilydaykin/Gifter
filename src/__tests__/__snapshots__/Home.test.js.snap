// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift categories are rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="categories"
      >
        <div
          class="category-poster category-poster--birthday"
        >
          <div
            class="category-poster__image"
            style="background-image: url(https://images.unsplash.com/photo-1530103862676-de8c9debad1d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8YmlydGhkYXl8ZW58MHx8MHx8&auto=format&fit=crop&w=800&q=60); background-position: center;"
          >
            <div
              class="category-poster__body"
            >
              <h2>
                Birthday
              </h2>
              <p>
                Browse Now
              </p>
            </div>
          </div>
        </div>
        <div
          class="category-poster category-poster--christmas"
        >
          <div
            class="category-poster__image"
            style="background-image: url(https://images.unsplash.com/photo-1451772741724-d20990422508?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80); background-position: center;"
          >
            <div
              class="category-poster__body"
            >
              <h2>
                Christmas
              </h2>
              <p>
                Browse Now
              </p>
            </div>
          </div>
        </div>
        <div
          class="category-poster category-poster--anniversary"
        >
          <div
            class="category-poster__image"
            style="background-image: url(https://images.unsplash.com/photo-1517837125937-53bd402f49d6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTR8fHZhbGVudGluZXxlbnwwfHwwfHw%3D&auto=format&fit=crop&w=800&q=60);"
          >
            <div
              class="category-poster__body"
            >
              <h2>
                Anniversary
              </h2>
              <p>
                Browse Now
              </p>
            </div>
          </div>
        </div>
        <div
          class="category-poster category-poster--thank you"
        >
          <div
            class="category-poster__image"
            style="background-image: url(https://images.unsplash.com/photo-1499744937866-d7e566a20a61?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTR8fHRoYW5rJTIweW91fGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=600);"
          >
            <div
              class="category-poster__body"
            >
              <h2>
                Thank You
              </h2>
              <p>
                Browse Now
              </p>
            </div>
          </div>
        </div>
        <div
          class="category-poster category-poster--wedding"
        >
          <div
            class="category-poster__image"
            style="background-image: url(https://images.unsplash.com/photo-1519225421980-715cb0215aed?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8d2VkZGluZ3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=800&q=60); background-position: center;"
          >
            <div
              class="category-poster__body"
            >
              <h2>
                Wedding
              </h2>
              <p>
                Browse Now
              </p>
            </div>
          </div>
        </div>
      </div>
      <footer
        class="footer"
      >
        <p
          class="footer__text"
        >
          Designed & Built by
           
          <a
            class="footer__link"
            href="/about"
          >
            Emily Daykin
          </a>
           
          ~ Copyright © 2022
        </p>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="categories"
    >
      <div
        class="category-poster category-poster--birthday"
      >
        <div
          class="category-poster__image"
          style="background-image: url(https://images.unsplash.com/photo-1530103862676-de8c9debad1d?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MXx8YmlydGhkYXl8ZW58MHx8MHx8&auto=format&fit=crop&w=800&q=60); background-position: center;"
        >
          <div
            class="category-poster__body"
          >
            <h2>
              Birthday
            </h2>
            <p>
              Browse Now
            </p>
          </div>
        </div>
      </div>
      <div
        class="category-poster category-poster--christmas"
      >
        <div
          class="category-poster__image"
          style="background-image: url(https://images.unsplash.com/photo-1451772741724-d20990422508?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80); background-position: center;"
        >
          <div
            class="category-poster__body"
          >
            <h2>
              Christmas
            </h2>
            <p>
              Browse Now
            </p>
          </div>
        </div>
      </div>
      <div
        class="category-poster category-poster--anniversary"
      >
        <div
          class="category-poster__image"
          style="background-image: url(https://images.unsplash.com/photo-1517837125937-53bd402f49d6?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTR8fHZhbGVudGluZXxlbnwwfHwwfHw%3D&auto=format&fit=crop&w=800&q=60);"
        >
          <div
            class="category-poster__body"
          >
            <h2>
              Anniversary
            </h2>
            <p>
              Browse Now
            </p>
          </div>
        </div>
      </div>
      <div
        class="category-poster category-poster--thank you"
      >
        <div
          class="category-poster__image"
          style="background-image: url(https://images.unsplash.com/photo-1499744937866-d7e566a20a61?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTR8fHRoYW5rJTIweW91fGVufDB8fDB8fA%3D%3D&auto=format&fit=crop&w=800&q=600);"
        >
          <div
            class="category-poster__body"
          >
            <h2>
              Thank You
            </h2>
            <p>
              Browse Now
            </p>
          </div>
        </div>
      </div>
      <div
        class="category-poster category-poster--wedding"
      >
        <div
          class="category-poster__image"
          style="background-image: url(https://images.unsplash.com/photo-1519225421980-715cb0215aed?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8N3x8d2VkZGluZ3xlbnwwfHwwfHw%3D&auto=format&fit=crop&w=800&q=60); background-position: center;"
        >
          <div
            class="category-poster__body"
          >
            <h2>
              Wedding
            </h2>
            <p>
              Browse Now
            </p>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      <p
        class="footer__text"
      >
        Designed & Built by
         
        <a
          class="footer__link"
          href="/about"
        >
          Emily Daykin
        </a>
         
        ~ Copyright © 2022
      </p>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
